services/navigation/builtin/builtin.go

```diff
@@ -399,14 +400,17 @@ func (svc *builtIn) startWaypoint(ctx context.Context, extra map[string]interfac
 			return svc.waypointReached(ctx)
 		}
 
-		// loop until no waypoints remaining
+		// do not exit loop - even if there are no waypoints
 		for {
+		WP:
 			if ctx.Err() != nil {
 				return
 			}
 
 			wp, err := svc.store.NextWaypoint(ctx)
-			if err != nil {
+			if err != nil && err.Error() == "no more waypoints" {
+				goto WP
```

Comment:
A `continue` should work here, no need for loop labels.

Reply:
I disagree that `continue` would be suffice.
My reasoning is:
suppose we have no remaining waypoints to navigate to.
the `NextWaypoint` method will then return `Waypoint{}, errNoMoreWaypoints`
if the program is allowed to `continue` we will eventually reach `navOnce` which expects a  context + waypoint
then inside `navOnce` we would do `wp.ToPoint(),` to give the destination to the moveonglobe call.
since `wp := Waypoint{}`, then `wp.ToPoint() = 0,0`
we do not want to navigate 0 0

Reply:
In order for that to happen, we would have to pass this check, so at that point `wp` would be valid.

`continue` will jump to a new iteration of the innermost loop. It doesn't mean "continue with the next line".

Reply:
> It doesn't mean "continue with the next line".

ai yayay. you're right, I'm sorry - not sure what I was thinking 🧠 💨